content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,290 @@
+
+The factoring will be progressive: we will start with a new repo that adds
+nothing to k/apiserver, and then progressively move generic functionality from
+kube-apiserver to the new repository. The new repo will be named
+k/generic-controlplane.
+  
+## CDI Support To The Device Plugin API
+
+CDI provides a standardized way of injecting complex devices into a container (i.e. devices that logically require more than just a single /dev node to be injected for them to work). This new feature enables plugin developers to utilize the CDIDevices field added to the CRI in 1.27 to pass CDI devices directly to CDI enabled runtimes (of which containerd and crio-o are in recent releases).
+    
+## Sidecar Containers Built-in Support
+
+This introduces a restartPolicy field to init containers and uses it to indicate that an init container is a sidecar container. Kubelet will start init containers with restartPolicy=Always in the order with other init containers, but instead of waiting for its completion, it will wait for the container startup completion.
+
+The condition for startup completion will be that the startup probe succeeded (or if no startup probe is defined) and postStart handler is completed. This condition is represented with the field Started of ContainerStatus type. See the section "Pod startup completed condition" for considerations on picking this signal.
+
+The field restartPolicy will only be accepted on init containers. The only supported value now is Always. No other values will be defined. Moreover, the field will be nullable so the default value will be "no value".
+
+Other values for restartPolicy of containers will not be accepted and containers will follow the logic currently implemented.
+
+Sidecar containers will not block Pod completion - if all regular containers are complete, sidecar containers will be terminated.
+
+During the sidecar startup stage, the restart behavior will be similar to init containers. If the Pod restartPolicy is Never, the sidecar container that failed during startup will NOT be restarted and the whole Pod will fail. If the Pod restartPolicy is Always or OnFailure, it will be restarted.
+
+Once the sidecar container is started (postStart completed and startup probe succeeded), these containers will be restarted even when the Pod restartPolicy is Never or OnFailure. Furthermore, sidecar containers will be restarted even during Pod termination.
+
+In order to minimize OOM kills of sidecar containers, the OOM adjustment for these containers will match or exceed the OOM score adjustment of regular containers in the Pod.
+
+This also enables sidecar containers (those will not be allowed for other init containers):
+
+- PostStart and PreStop lifecycle handlers for sidecar containers
+- All probes (startup, readiness, liveness)
+    
+Readiness probes of sidecars will contribute to determining the whole Pod readiness.
```

Comment:
We should just link to the docs (once they exist) rather than give more detail here.

Reply:
> We should just link to the docs (once they exist) rather than give more detail here.

What's happening on this?

Reply:
We can point to this: https://github.com/kubernetes/website/blob/dev-1.28/content/en/docs/concepts/workloads/pods/init-containers.md

Reply:
Thanks done.